apps/gatsby/src/Sidebar.js

```diff
@@ -42,8 +41,68 @@ export default class Sidebar extends React.Component {
     this.debounceInterval = setInterval(this.refreshPreview, 1000);
   };
 
-  componentDidMount() {
+  // Recursive helper to return slug values buried in a chain of references
+  resolveReferenceChain = async (sdk, array, index, parentId) => {
+    // Full entry to access child fields
+    const fullParentEntry = await sdk.space.getEntry(parentId)
+    // Child field
+    const childField = fullParentEntry.fields[array[index + 1]][sdk.locales.default]
+    if (Array.isArray(childField)) {
+    console.error("Gatsby Preview App: You are trying to search for a slug in a multi reference field. Only single reference fields are searchable with this app. Either change the field to a single reference, or change the field you are searching for in the slug.")
+      return ""
+    }
+    if (index + 2 < array.length) {
+       return this.resolveReferenceChain(sdk, array, (index + 1), childField.sys.id)
+    } else {
+      return childField
+    }
+  }
+
+  buildSlug = async () => {
```

Comment:
Slug should be rebuilt on value change, if the reference is a field (otherwise I see one thing in the editor, but the Gatsby button will redirect me to the old url)

Reply:
I opted to rerun `buildSlug` in with `entry.onSysChanged` event listener callback. It updates too often, but it will also be far more likely to rebuild when a slug buried in a reference is changed, and it should definitely update every time a slug field directly on an entry is updated.